homeassistant/components/graphite/__init__.py

```diff
@@ -31,6 +33,9 @@
             {
                 vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
                 vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                vol.Optional(CONF_PROTOCOL, default=DEFAULT_PROTOCOL): cv.matches_regex(
```

Comment:
Instead of using a regex, does it make sense to directly constrain these values with something like `vol.Any("tcp", "udp")`? Similar to https://github.com/home-assistant/core/blob/dev/homeassistant/components/ecobee/climate.py#L130

Reply:
Agree. It should be faster too.